frame/assets/src/lib.rs

```diff
@@ -747,14 +773,17 @@ pub mod pallet {
 				if asset == T::NativeAssetId::get() {
 					return <<T as Config>::NativeCurrency>::set_balance(who, amount)
 				}
+				ValidatedAsset::<T>::new(asset).map_err(|_| Error::<T>::InvalidCurrency)?;
```

Comment:
@PoisonPhang @benluelo is this better?

Reply:
Error handling looks better, but this still isn't correctly using Validated.
Something more like this is how Validated is intended to be used
```rust
// snip...
			fn set_balance(
				asset: Validated<Self::AssetId, ValidatedAsset>,
				who: &T::AccountId,
				amount: Self::Balance,
			) -> DispatchResult {
                let asset = asset.value;
				if asset == T::NativeAssetId::get() {
					return <<T as Config>::NativeCurrency>::set_balance(who, amount)
				}
                <<T as Config>::MultiCurrency>::set_balance(asset, who, amount)
             }
/ snip...
```
Again, see [this example](https://github.com/ComposableFi/composable/blob/main/frame/vault/src/lib.rs#L476) for the `amount` parameter from Vault

Reply:
You may find the docs in [validation.rs](https://github.com/ComposableFi/composable/blob/main/frame/composable-support/src/validation.rs) helpful